src/Microsoft.AspNetCore.Authentication2/AuthenticationSchemeBuilder.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Authentication2
+{
+    public class AuthenticationSchemeBuilder
+    {
+        public AuthenticationSchemeBuilder(string name)
+        {
+            Name = name;
+        }
+
+        public string Name { get; }
+
+        public Type HandlerType { get; set; }
+
+        // Holds things like the configured options instances for the handler
```

Comment:
Tradeoff vs generic hell, once I implement a few more auth handlers, I'll do a cleanup/rollup pass to see if we need anything at this layer but there's is an equivalent `AuthenticationSchemeHandler<TOptions>` that is similar to `AuthenticationHandler<TOptions>` today.  The options don't need to be typed at the scheme builder layer, since these untyped options are only used by the scheme handler instance later, when it can just cast it to the type it knows.  Its not related to the global singleton AuthenticationOptions which deals only with the various schemes and handler types.